### ISSUE TITLE
Proposed Bind Commands break move up/down feature in non-JSON files

### ISSUE THREAD

**Issue by wojtekmaj** [zardoy/vscode-move-statement#1]
If you configure Bind Commands to use alt+up/alt+down keybindings, it breaks move up/down in non-JSON files (e.g. JavaScript React) as this extension doesn't seem to support it, and the default action is not triggered.

According to VSCode documentation, you should be able to write `when` like: `"editorTextFocus && resourceLangId == 'json' || resourceLangId == 'jsonc'"`

**Comment by zardoy** [comment]
Sorry for the delay. Actually was online, but it takes some time for me to see the notification.

> If you configure Bind Commands to use alt+up/alt+down keybindings, it breaks move up/down in non-JSON files

There is no way of happening that, I'd be happy to dig in.

According to VSCode documentation, you should be able to write `when` like: `editorTextFocus && editorLangId == 'json' || editorLangId == 'jsonc'` or the same but shorter: `editorTextFocus && editorLangId =~ /json|jsonc/`. *Don't use `resourceLangId`* use `editorLangId` instead as `resourceLangId` refers to file selected in explorer area (it can be different if you open file outside of current workspace and it won't receive selection). That's why `editorLangId` makes more sense to use with connection with `editorTextFocus`

> it breaks move up/down in non-JSON files (e.g. JavaScript React) as this extension doesn't seem to support it, and the default action is not triggered

Can you elaborate and show me some examples? I didn't have a chance to test it widely but would appreciate additional feedback e.g. examples were it fails. Maybe you're trying to mix different kind of statements? Please show me.

**Comment by wojtekmaj** [comment]
You're totally right on the `editorLangId` thing. And thanks for reminding me of the possibility of using regex!

> There is no way of happening that, I'd be happy to dig in.

> Can you elaborate and show me some examples? I didn't have a chance to test it widely but would appreciate additional feedback e.g. examples were it fails. Maybe you're trying to mix different kind of statements? Please show me.

Nothing fancy. It breaks for me on simple import list.

Here's a video. First, I use the extension without  `&& editorLangId =~ /json|jsonc/` bit, then - with. Sound on, intentional :D

https://user-images.githubusercontent.com/5426427/206392059-08a95650-eeda-4a12-b212-4a34eb3b7c4d.mov

**Comment by zardoy** [comment]
Thanks for the details! I just watched video and I see that everything works as expected. alt+down in json executes this extension command (as you configured) and alt+down in other files is builtin [keybinding](https://github.com/microsoft/vscode/blob/3b1c78409acec77f2c1ba89b105e34d757a987cd/src/vs/editor/contrib/linesOperations/browser/linesOperations.ts#L223). You can try to comment your keybindings - the result should be same.

<img width="700" alt="Screenshot 2022-12-08 at 14 01 01" src="https://user-images.githubusercontent.com/46503702/206430423-9b630a9a-e15d-467a-9367-d808a29aadec.png">


And yes, extension's command doesn't work on import statements (last part in breadcrumbs indicates the statement that is going to be moved, `...` means there is no statement in outline in current position), but they should work well on other statements until there're properly formatted.
